front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/plugininfo/PluginInfoService.java

```diff
@@ -53,7 +54,19 @@ public PluginInfo upsert(@Nonnull PluginInfo pluginInfo) {
     validate(pluginInfo);
 
     try {
-      repository.findById(pluginInfo.getId());
+      PluginInfo currentPluginInfo = repository.findById(pluginInfo.getId());
+      List<PluginInfo.Release> newReleases = new ArrayList<>(pluginInfo.getReleases());
+      List<PluginInfo.Release> oldReleases = new ArrayList<>(currentPluginInfo.getReleases());
+      new ArrayList<>(newReleases)
+          .forEach(
+              release -> { // remove it from the old release list if it is being updated.
+                oldReleases.removeIf(
+                    oldRelease -> oldRelease.getVersion().equals(release.getVersion()));
+              });
+
+      List<PluginInfo.Release> allReleases = new ArrayList<>();
+      Stream.of(oldReleases, newReleases).forEach(allReleases::addAll);
+      pluginInfo.setReleases(allReleases);
```

Comment:
I'm not certain we want to allow updating previous releases - in my mind these are immutable: We wouldn't want someone to come along and update an old release to point to a new artifact, for example. Shouldn't the upsert operation just add new releases to the existing releases list? If they're trying to update a previous release, I'd have expected an error. Happy to be wrong, here, though.

Reply:
if it does need to happy to fix a broken publication or something it would be best to be explicit (deleting an old version first, etc)

Reply:
Yh.. Add the new release to the existing release is what I started with, but I was thinking about other attributes on the release say for eg `required` restrictions etc which I thought we can update. May be..?  Not a usecase to worry about now .. I will just add the new release to the current release .